oeqLinkMigrationLTI/src/main/java/org/apereo/openequella/integration/blackboard/linkmigrationlti/Fixer.java

```diff
@@ -0,0 +1,406 @@
+			  logMessage(0, "An error occurred: " + e.getMessage());
+			  logMessage(0, "See Blackboard service logs for details");
+			  bbLogs.logError("An error occurred trying to fix EQUELLA content links", e);
+			}
+		  }
+		};
+		new Thread(runner, "Link Fixer Thread").start();
+	  }
+	}
+  }
+
+  protected void recurseContent(ContentDbLoader contentLoader, CourseToc courseToc, BbList contentList, String placementExternalId, int level)
+	throws Exception {
+	for (int j = 0; j < contentList.size(); j++) {
+
+	  Content content = (Content) contentList.get(j);
+	  lookedAt++;
+
+	  String handler = content.getContentHandler();
+
+	  boolean tleResource = handler.equals("resource/tle-resource") || handler.equals("resource/tle-myitem")
+		|| handler.equals("resource/tle-plan");
+	  if (tleResource) {
+
+		logMessage(level, "Title: " + content.getTitle());
+		try {
+		  logMessage(level, "getDataType().getName(): " + content.getDataType().getName());
+		} catch (Exception exc) {
+		  logMessage(1, "Error with getDataType: " + exc.getMessage());
+		}
+
+		ExtendedData extendedData2 = content.getExtendedData();
+		if (extendedData2 != null) {
+		  mapToString(extendedData2.getValues(), level, "Old Extended Data");
+		}
+	  }
+
+	  if (tleResource) {
+		logMessage(0, "Updating:  " + contentDisplay(content));
+		try {
+		  fixContent(courseToc, content, level, placementExternalId);
+		  logMessage(0, "Success");
+		} catch (Exception ex) {
+		  logMessage(0, "Updating ERROR: " + ex.getMessage());
+		  for (StackTraceElement element : ex.getStackTrace()) {
+			logMessage(0, element.toString());
+		  }
+		}
+	  }
+	  recurseContent(contentLoader, courseToc, getChildren(contentLoader, content.getId()), placementExternalId, level + 1);
+	}
+  }
+
+  private void mapToString(Map<String, String> map, int level, String label) {
+	try {
+	  Iterator<String> i = map.keySet().iterator();
+	  while (i.hasNext()) {
+		String attribute = i.next();
+		logMessage(level, attribute + " : " + map.get(attribute));
+	  }
+	} catch (Exception exc) {
+	  logMessage(level, label + " error : " + exc.getMessage());
+	}
+  }
+
+  protected BbList getChildren(ContentDbLoader contentLoader, Id bbContentId) throws PersistenceException {
+	return contentLoader.loadChildren(bbContentId);
+  }
+
+  private void fixContent(CourseToc courseToc, Content ltiContent, int level, String placementExternalId) throws Exception {
+	equellaLookedAt++;
+
+	ExtendedData extendedData = ltiContent.getExtendedData();
+	//Update the URL
+	String newUrl = getEquellaUrl() + extendedData.getValue("url");
+	logMessage(level, "NewUrl: " + newUrl);
+	ltiContent.setUrl(newUrl);
+	//Update the Host
+	String newUrlHost = getDomainName(getEquellaUrl());
+	logMessage(level, "NewUrlHost: " + newUrlHost);
+	ltiContent.getBbAttributes().setString("UrlHost", newUrlHost);
+	//Update the Type
+	ltiContent.setContentHandler("resource/x-bb-blti-link");
```

Comment:
doesn't the Bb LTI Provider placement offer a configuration for the content handle?  Or is this something different?

Reply:
This is the content handler that is generated by the new links that our new code is creating. It seems to be the needed content handler for all the bb lti links. So to convert the old links to new links, I changed it.